jenkins/test/testsuite/Jenkinsfile

```diff
@@ -68,20 +70,46 @@ pipeline {
             post() {
                 always {
                     script {
-                        publish()
+                        publishResults()
                     }
                     cleanWs disableDeferredWipeout: true, deleteDirs: true
                 }
             }
         }
     }
+    post() {
+        success {
+            node('Jenkins-Agent-al2-x64-m5xlarge') {
+                script {
+                    def stashed = lib.jenkins.Messages.new(this).get(['Integ-Test'])
+                    publishNotification(":white_check_mark:", "Successful Build", "\n${stashed}")
+                }
+            }
+        }
+        failure {
+            node('Jenkins-Agent-al2-x64-m5xlarge') {
+                script  {
+                    def stashed = lib.jenkins.Messages.new(this).get(['Integ-Test'])
+                    publishNotification(":warning:", "Failed Build", "\n${stashed}")
+                }
+            }
+        }
+    }
 }
 
 /** Publishes test results to S3 **/
-void publish() {
+void publishResults() {
```

Comment:
Take a look at https://github.com/opensearch-project/opensearch-build/pull/979.

We've introduced https://github.com/opensearch-project/opensearch-build/blob/main/vars/uploadArtifacts.groovy, which uploads `builds` and `dist` and is identical to this entire piece of code. Extract the list of folders to upload and pass in `test-results` here, and `["builds", "dist"]` in the other Jenkinsfile(s).

Reply:
Used the same functions for test JenkinsFile as well.